libvmi/driver/kvm/kvm.c

```diff
@@ -179,6 +186,51 @@ parse_reg_value(
     }
 }
 
+static reg_t
+parse_seg_reg_value(
+    char *regname,
+    char *ir_output,
+    int type)
+{
+    int offset;
+    int size;
+
+    if (NULL == ir_output || NULL == regname) {
+        return 0;
+    }
+
+    switch(type) {
+        case SEGMENT_SELECTOR:
+        offset = 4;
+        size = 4;
+        break;
+        case SEGMENT_BASE:
+        offset = 9;
+        size = 16;
+        break;
+        case SEGMENT_LIMIT:
+        offset = 26;
+        size = 8;
+        break;
+        case SEGMENT_ATTR:
+        offset = 35;
+        size = 8;
+        break;
+        default:
+        return 0;
+    }
+
+    char *ptr = strcasestr(ir_output, regname);
+
+    if (NULL != ptr) {
+        ptr += offset;
+        return (reg_t) strtoll(ptr, (char **) NULL, size);
```

Comment:
The strtoll signature is as follows
long long int strtoll(const char *nptr, char **endptr, int base);
So the third arg of strtoll must always be 16